src/web/handlers.rs

```diff
@@ -355,7 +355,15 @@ pub async fn post_collection_batch(
     match result {
         Ok(_) => success.extend(bso_ids),
         Err(e) if e.is_conflict() => return Err(e.into()),
-        Err(_) => failed.extend(bso_ids.into_iter().map(|id| (id, "db error".to_owned()))),
+        Err(apperr) => match apperr.kind() {
+            // If we're over quota, return immediately with a 403 to let the client know.
+            // Otherwise the client will simply keep retrying records.
+            ApiErrorKind::Db(dberr) => match dberr.kind() {
+                DbErrorKind::Quota => return Ok(apperr.into()),
```

Comment:
lol @ "return Ok(apperr)"

Thanks actix-web!

Reply:
heh, not actix-web, that was me.Trying to make things a bit clearer.